merkle/src/lib.rs

```diff
@@ -46,21 +74,68 @@ impl<F: Field, H: Hasher<F>> MerkleTree<F, H> {
     /// # Returns
     ///
     /// Returns an error type for easier "monadic style" chaining.
-    #[allow(dead_code)]
-    #[allow(clippy::result_unit_err)]
     pub fn verify_decommitment(
         &self,
         comm: H::Output,
         to_verify: &[(usize, H::Output)],
-        siblings: impl Iterator<Item = H::Output>,
-    ) -> Result<(), ()> {
-        if self.root_decommitment(to_verify, siblings).ok_or(())? == comm {
-            Ok(())
+    ) -> Result<bool, MerkleError> {
+        assert!(!to_verify.is_empty(), "Empty verify queries");
+
+        let queries_idx: Vec<usize> = to_verify.iter().map(|&(x, _)| x).collect();
+        let siblings = self
```

Comment:
The verifier needs to be able to read these as a stream of siblings out of a proof transcript.